src/ray/core_worker/task_manager.cc

```diff
@@ -413,7 +444,7 @@ void TaskManager::DelObjectRefStream(const ObjectID &generator_id) {
   for (const auto &object_id : object_ids_unconsumed) {
     std::vector<ObjectID> deleted;
     reference_counter_->RemoveLocalReference(object_id, &deleted);
-    RAY_CHECK_EQ(deleted.size(), 1UL);
+    RAY_CHECK_GE(deleted.size(), 1UL);
```

Comment:
This is unrelated, but this is other issue I found. When a return reference includes  a nested reference, this could be bigger than 1.